src/modules/tests/test-data/cat-b/test-data.cat-b.selector.ts

```diff
@@ -0,0 +1,100 @@
+
+export const getTellMeQuestion = (state: CatBUniqueTypes.VehicleChecks): VehicleChecksQuestion => {
+  const questionProvider: QuestionProvider =  new QuestionProvider();
+  return questionProvider
+    .getTellMeQuestions(TestCategory.B)
+    .find(question => question.code === get(state, 'tellMeQuestion.code'));
+};
+
+export const isTellMeQuestionSelected = (
+  state: CatBUniqueTypes.VehicleChecks) => get(state, 'tellMeQuestion.code') !== undefined;
+
+export const isTellMeQuestionCorrect =
+  (state: CatBUniqueTypes.VehicleChecks) => get(state, 'tellMeQuestion.outcome') === CompetencyOutcome.P;
+
+export const isTellMeQuestionDrivingFault = (state: CatBUniqueTypes.VehicleChecks) =>
+  get(state, 'tellMeQuestion.outcome') === CompetencyOutcome.DF;
+
+export const tellMeQuestionOutcome = (state: CatBUniqueTypes.VehicleChecks) => get(state, 'tellMeQuestion.outcome');
+
+export const getSelectedTellMeQuestionText = (state: CatBUniqueTypes.VehicleChecks) => {
+  const questionProvider: QuestionProvider = new QuestionProvider();
+
+  const tellMeQuestionText = questionProvider
+    .getTellMeQuestions(TestCategory.B)
+    .find(question => question.code === get(state, 'tellMeQuestion.code'));
+
+  if (!tellMeQuestionText) {
+    return '';
+  }
+  return `${get(state, 'tellMeQuestion.code')} - ${tellMeQuestionText.shortName}`;
+};
+
+export const getShowMeQuestion = (state: CatBUniqueTypes.VehicleChecks) => {
+  const questionProvider: QuestionProvider = new QuestionProvider();
+  return questionProvider
+    .getShowMeQuestions(TestCategory.B)
+    .find(question => question.code === get(state, 'showMeQuestion.code'));
+};
+
+export const getCatBLegalRequirements = (data: CatBUniqueTypes.TestData): CatBLegalRequirements => {
+  return {
+    normalStart1: data.testRequirements.normalStart1 || false,
+    normalStart2: data.testRequirements.normalStart2 || false,
+    angledStart: data.testRequirements.angledStart || false,
+    hillStart: data.testRequirements.hillStart || false,
+    manoeuvre: hasManoeuvreBeenCompleted(data) || false,
+    vehicleChecks: hasVehicleChecksBeenCompleted(data) || false,
+    eco: data.eco.completed || false,
+  };
+};
+
+export const hasVehicleChecksBeenCompleted = (data: CatBUniqueTypes.TestData): boolean => {
+  const showMeQuestionOutcome = data.vehicleChecks.showMeQuestion.outcome;
+  const tellMeQuestionOutcome = data.vehicleChecks.tellMeQuestion.outcome;
+
+  return (
+      showMeQuestionOutcome === CompetencyOutcome.P ||
+      showMeQuestionOutcome === CompetencyOutcome.DF ||
+      showMeQuestionOutcome === CompetencyOutcome.S ||
+      showMeQuestionOutcome === CompetencyOutcome.D)
+      && (
+        tellMeQuestionOutcome === CompetencyOutcome.P ||
+        tellMeQuestionOutcome === CompetencyOutcome.DF ||
+        tellMeQuestionOutcome === CompetencyOutcome.S ||
+        tellMeQuestionOutcome === CompetencyOutcome.D
+      );
```

Comment:
I think the outcome might be typed so that the only possible outcomes are P, DF, S, D, 
so i think just checking that outcome != null for showMeQuestion and TellMeQuestion might be enough.

Reply:
Done